### PR TITLE
Use larger partition scheme, add check_tool, move identical entries to global env

### PR DESCRIPTION
**Pull request by fiendie**

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -4,6 +4,14 @@ src_dir = src
 extra_configs = platformio_extra.ini
 
 [env]
+platform = espressif32 @^5
+board = az-delivery-devkit-v4
+board_build.filesystem = littlefs
+board_build.partitions = huge_app.csv
```
Comment (kjyv) [inline platformio.ini:10]: I think huge_app doesn't allow ota anymore

Comment (kjyv) [reply platformio.ini:10]: Compared to https://github.com/espressif/arduino-esp32/blob/master/tools/partitions/no_ota.csv, the code partition is a bit smaller and spiffs is larger

Comment (kjyv) [reply platformio.ini:10]: I've more or less maximized the filesystem in PR #326 already

Comment (fiendie) [reply platformio.ini:10]: default.csv then, partitions_4M.csv is a partition scheme part of Espressif IDF not the Arduino framework.

Comment (kjyv) [reply platformio.ini]: It's a file that's in the PR

Comment (fiendie) [reply platformio.ini]: Either way, it works fine with default.csv, on PR #326 as well.

**Pull request merged**